### PR TITLE
Add project metadata to PureModelContext in File Generation plugin

### PR DESCRIPTION
**Pull request by tomwilson-gs**
Adds the SDLC project groupId/artifactId/version to the SDLC metadata in the built PureModelContext for downstream consumption in ArtifactGenerationExtensions.

### CODE REVIEW

**Review by aormerod-gs — COMMENTED** [review]
Is there any unit test we can add? (To confirm that if mavenProject is injected that it gets propagated to the SDLC info)

**Review by aormerod-gs — APPROVED** [review]

File: legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java [hunk 1]
```diff
@@ -286,6 +292,18 @@ private static ResolvedPackageableElementFilter resolvePackageableElementFilter(
...
     }
 
+    private SDLC buildSDLCInfo()
+    {
+        MavenProject projectToInspect = mavenProject.getParent() != null ? mavenProject.getParent() : mavenProject;
+
+        AlloySDLC sdlcInfo = new AlloySDLC();
+        sdlcInfo.groupId = projectToInspect.getGroupId();
+        sdlcInfo.artifactId = projectToInspect.getArtifactId();
+        sdlcInfo.version = projectToInspect.getVersion();
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java:302]: You should be able to get the version from the project directly without needing to go to the parent.

File: legend-sdlc-language-pure-compiler/src/main/java/org/finos/legend/sdlc/language/pure/compiler/toPureGraph/PureModelBuilder.java [hunk 2]
```diff
@@ -135,6 +136,12 @@ public PureModelBuilder withEntitiesIfPossible(Entity... entities)
         return this;
     }
 
+    public PureModelBuilder withSDLC(SDLC sdlc)
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-language-pure-compiler/src/main/java/org/finos/legend/sdlc/language/pure/compiler/toPureGraph/PureModelBuilder.java:139]: Should we also add builder methods for setting the protocol?

**Review by kevin-m-knight-gs — CHANGES_REQUESTED** [review]
I think there's one minor change needed for getting group and artifact ids. Also, should we consider setting the protocol along with the SDLC?

File: legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java (hunk 1 above)
Comment (tomwilson-gs) [reply legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java]: Agreed, amended

File: legend-sdlc-language-pure-compiler/src/main/java/org/finos/legend/sdlc/language/pure/compiler/toPureGraph/PureModelBuilder.java (hunk 2 above)
Comment (tomwilson-gs) [reply legend-sdlc-language-pure-compiler/src/main/java/org/finos/legend/sdlc/language/pure/compiler/toPureGraph/PureModelBuilder.java:140]: Added this

File: legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java [hunk 3]
```diff
@@ -286,6 +292,18 @@ private static ResolvedPackageableElementFilter resolvePackageableElementFilter(
         return new ResolvedPackageableElementFilter(resolvedElementsByPath, elementFilter.packages);
     }
 
+    private SDLC buildSDLCInfo()
+    {
+        MavenProject projectToInspect = mavenProject.getParent() != null ? mavenProject.getParent() : mavenProject;
+
+        AlloySDLC sdlcInfo = new AlloySDLC();
+        sdlcInfo.groupId = projectToInspect.getGroupId();
+        sdlcInfo.artifactId = projectToInspect.getArtifactId();
```
[the remark this answers is not in the document]
Comment (tomwilson-gs) [reply legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java]: Thanks for the feedback - I've refactored to pull the groupId and artifactId from the project.json file.

File: legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java [hunk 4]
```diff
@@ -286,6 +297,45 @@ private static ResolvedPackageableElementFilter resolvePackageableElementFilter(
...
+            ObjectMapper mapper = ObjectMapperFactory.getNewStandardObjectMapper();
+            ReducedProjectConfiguration projectConfiguration = mapper.readValue(stream, ReducedProjectConfiguration.class);
+
+            sdlcInfo.groupId = projectConfiguration.getGroupId();
+            sdlcInfo.artifactId = projectConfiguration.getArtifactId();
+            sdlcInfo.version = mavenProject.getVersion();
+        }
+        catch (IOException e)
+        {
+            getLog().warn("Unable to build SDLC info");
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java:318]:
```suggestion
            getLog().warn("Unable to build SDLC info", e);
```

File: legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/ReducedProjectConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.finos.legend.sdlc.domain.model.project.configuration.MetamodelDependency;
+import org.finos.legend.sdlc.domain.model.project.configuration.ProjectConfiguration;
+import org.finos.legend.sdlc.domain.model.project.configuration.ProjectDependency;
+import org.finos.legend.sdlc.domain.model.project.configuration.ProjectStructureVersion;
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ReducedProjectConfiguration implements ProjectConfiguration
```
Comment (kevin-m-knight-gs) [inline legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/ReducedProjectConfiguration.java:27]: I don't think there's a need for this to be public.
```suggestion
class ReducedProjectConfiguration implements ProjectConfiguration
```

**Review by kevin-m-knight-gs — CHANGES_REQUESTED** [review]
Just two minor comments

**Review by kevin-m-knight-gs — APPROVED** [review]

**Pull request merged**